README.rst

```diff
@@ -165,6 +165,20 @@ providers package. The localization of Faker is an ongoing process, for
 which we need your help. Please don't hesitate to create a localized
 provider for your own locale and submit a Pull Request (PR).
 
+Optimizations
+-------------
+The Faker constructor takes two performance-related arguments. One is
+``use_weighting``, which specifies whether to attempt to have the frequency
+of values match real-world frequencies (e.g. the English name Gary would
+be much more frequent than the name Lorimer). If ``use_weighting`` is ``False``,
+then all items have an equal chance of being selected, and the selection
+process is much faster. The default is ``True``.
+
+The second performance option is ``use_internal_caches`` which does not change the
```

Comment:
What would be a good reason for reason to turn this off? I'm wondering if we could make it simpler and just always cache.

Reply:
Really just about space/time tradeoff. I'm pretty fuzzy about what the normal and worst-case memory usage of Faker looks like. Like if you load every provider in every locale...how much RAM does it take and then if we increase that by a certain percent by caching.

It's up to you: I don't feel strongly one way or the other. Just a way of covering my butt in case someone says that their Faker memory usage spiked after upgrading...

Reply:
I dont think users will load up that many locales at once. I would rather keep it simple, and add this knob later if it turns out to be too memory-heavy.